File: C++/easy/1_two_sum.cc
#include <stdint.h>
#include <vector>

class Solution
{
public:
    std::vector<int> twoSum(std::vector<int> &nums, int target)
    {
        std::vector<int> solution(2);

        for (size_t i = 0; i < nums.size(); i++)
        {
            for (size_t j = i + 1; j < nums.size(); j++)
            {
                if (nums[i] + nums[j] == target)
                {
                    solution[0] = i;
                    solution[1] = j;
                    return solution;
                }
            }
        }

        return solution;
    }
};

File: Python/easy/1_two_sum.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen ={}
        for (idx, num) in enumerate(nums):
            remainder = target - num
            rIdx = seen.get(remainder)
            if rIdx is None:
                seen[num] = idx
                continue

            return [rIdx, idx]


        return []

